src/Auth0.AuthenticationApi/AuthenticationApiClient.cs

```diff
@@ -211,134 +196,118 @@ public async Task<AccessTokenResponse> GetTokenAsync(AuthorizationCodePkceTokenR
         /// Request an Access Token using the Client Credentials Grant flow.
         /// </summary>
         /// <param name="request">The <see cref="ClientCredentialsTokenRequest"/> containing the information of the request.</param>
-        /// <returns>An <see cref="AccessTokenResponse"/> containing the token information</returns>
+        /// <returns><see cref="AccessTokenResponse"/> containing the desired token information.</returns>
         public Task<AccessTokenResponse> GetTokenAsync(ClientCredentialsTokenRequest request)
         {
-            return Connection.PostAsync<AccessTokenResponse>("oauth/token", null, new Dictionary<string, object>
-                {
-                    {"grant_type", "client_credentials"},
-                    {"client_id", request.ClientId},
-                    {"client_secret", request.ClientSecret},
-                    {"audience", request.Audience}
-                },
-                null,
-                null,
-                null,
-                null);
+            var body = new Dictionary<string, string> {
+                { "grant_type", "client_credentials" },
+                { "client_id", request.ClientId },
+                { "client_secret", request.ClientSecret },
+                { "audience", request.Audience } };
+
+            return connection.SendAsync<AccessTokenResponse>(
+                HttpMethod.Post,
+                tokenUri,
+                body);
         }
 
         /// <summary>
-        /// Given a <see cref="RefreshTokenRequest"/>, it will retrieve a refreshed access token from the authorization server.
+        /// Refresh all tokens using the Refresh Token obtained during authorization.
         /// </summary>
-        /// <param name="request">The refresh token request details, containing a valid refresh token.</param>
-        /// <returns>The new token issued by the server.</returns>
+        /// <param name="request"><see cref="RefreshTokenRequest"/> containing Refresh Token and associated parameters.</param>
+        /// <returns><see cref="AccessTokenResponse"/> containing the desired token information.</returns>
         public async Task<AccessTokenResponse> GetTokenAsync(RefreshTokenRequest request)
         {
-            var parameters = new Dictionary<string, object> {
+            var body = new Dictionary<string, string> {
                 { "grant_type", "refresh_token" },
-                { "refresh_token", request.RefreshToken },
                 { "client_id", request.ClientId },
-                { "client_secret", request.ClientSecret }
+                { "client_secret", request.ClientSecret },
+                { "refresh_token", request.RefreshToken }
             };
 
-            if (!string.IsNullOrEmpty(request.Audience))
-            {
-                parameters.Add("audience", request.Audience);
-            }
+            body.AddIfNotEmpty("audience", request.Audience);
+            body.AddIfNotEmpty("scope", request.Scope);
```

Comment:
You never need to send these. The refresh token is tied to a specific audience and scope:

https://auth0.com/docs/api/authentication#refresh-token

Reply:
Okay, I could remove them but that's going beyond the scope of this PR.

Reply:
It looks like you're re-writing quite a few of these methods and this could be part of that cleanup. Sending or not sending does not change the behavior of this endpoint, it would be like sending a randomly-generated parameter and value. 

Up to you, it won't hold up my approval.

Reply:
I'd rather do it as a separate PR - that way I can refer to the change separately in the change log. From an outside perspective they are two different things.